doc/developer-guides/hld/hv-rdt.rst

```diff
@@ -0,0 +1,54 @@
+.. _hv_rdt:
+
+RDT Allocation Feature Supported by Hypervisor
+##############################################
+
+The hypervisor allows to use RDT (Resource Director Technology) allocation features to optimize performances of VMs. There are 2 sub-features: CAT (Cache Allocation Technology) and MBA(Memory Bandwidth Allocation), CAT is for cache resources and MBA is for memory bandwidth resources. The RDT in hypervisor is integrated with VM-Configuration feature, the resources consumed by each VM can be configured and restricted at compile time.
+
+CAT High-Level Design
+*********************
+
+Introduction of CAT capabilities
+===============================
+
+On a platform which support CAT, each CPU can mask last-level-cache (LLC) with a cache mask, the masked cache ways cannot be evicted by this CPU. In terms of SDM, CAT abilities are enumerated via CPUID, and configured via MSR registers, these are:
+
+* CPUID.10H contains CAT capabilities, such as cache mask type(L2/L3), mask bit
+  length, number of masks.
+
+* Cache masks are set aside in IA32_type_MASK_n, each one of these MSRs can
+  hold a cache mask. They are shared by CPUs who share the same LLC; CPU set
+  the class-of-service (CLOS) of IA32_type_MASK_n in its IA32_PQR_ASSOC MSR, to
```

Comment:
CPU sets the RMID (CLOS ID) field of its IA32_PQR_ASSOC MSR to select a cache mask to take effect.

Reply:
Use this!